agents_infra/agents_infra/agents/base.py

```diff
@@ -540,7 +474,7 @@ def post_data(
             **kwargs: Key-value pairs to be appended to the header.
         """
         if to_controller:
-            if not self.controller_url:
+            if not self.config.url:
```

Comment:
Why do you use `config.get(<attribite_name>)` in some cases and then just `config.<attribute_name>` in other cases?